src/allocated_nonnative_field_var.rs

```diff
@@ -754,59 +855,11 @@ impl<TargetField: PrimeField, BaseField: PrimeField> AllocVar<TargetField, BaseF
     ) -> R1CSResult<Self> {
         let ns = cs.into();
         let cs = ns.cs();
-
-        let optimization_type = match cs.optimization_goal() {
-            OptimizationGoal::None => OptimizationType::Constraints,
-            OptimizationGoal::Constraints => OptimizationType::Constraints,
-            OptimizationGoal::Weight => OptimizationType::Weight,
-        };
-
-        let params = get_params(
-            TargetField::size_in_bits(),
-            BaseField::size_in_bits(),
-            optimization_type,
-        );
-        let zero = TargetField::zero();
-
-        let elem = match f() {
-            Ok(t) => *(t.borrow()),
-            Err(_) => zero,
-        };
-        let elem_representations = Self::get_limbs_representations(&elem, optimization_type)?;
-        let mut limbs = Vec::new();
-
-        for limb in elem_representations.iter() {
-            limbs.push(FpVar::<BaseField>::new_variable(
-                ark_relations::ns!(cs, "alloc"),
-                || Ok(limb),
-                mode,
-            )?);
-        }
-
-        let num_of_additions_over_normal_form = if mode != AllocationMode::Witness {
-            BaseField::zero()
-        } else {
-            BaseField::one()
-        };
-
+        let this = Self::new_variable_unchecked(ns!(cs, "alloc"), f, AllocationMode::Witness)?;
```

Comment:
Hmm this always allocates in witness mode, right? Should it be taking `mode` as input instead?

Reply:
@alex-ozdemir ping on this

Reply:
```suggestion
        let this = Self::new_variable_unchecked(ns!(cs, "alloc"), f, mode)?;
```